ccv/utils.py

```diff
@@ -1,7 +1,47 @@
 # -*- coding: utf-8 -*-
 
-import re
+from collections import defaultdict
 
 
 def normalize_string(s: str) -> str:
+    """
+    :param s:
+    :return:
+    """
     return s.strip()
+
+
+def parse_integer(s: str) -> int or None:
+    """
+    :param s:
+    :return:
+    """
+    int(s) if isinstance(s, str) and len(s) > 0 else None
+
+
+def etree_to_dict(t) -> dict:
+    """
+    Converts the xml tree to python dictionary
+    :param t: xml root element
+    :return: python dictionary
+    """
+    d = {t.tag: {} if t.attrib else None}
+    children = list(t)
+    if children:
+        dd = defaultdict(list)
+        for dc in map(etree_to_dict, children):
+            for k, v in dc.items():
+                dd[k].append(v)
+        d = {t.tag: {k: v[0] if len(v) == 1 else v
+                     for k, v in dd.items()}}
+    if t.attrib:
+        d[t.tag].update(('' + k, v)
+                        for k, v in t.attrib.items())
+    if t.text:
+        text = t.text.strip()
+        if children or t.attrib:
+            if text:
+                d[t.tag]['text'] = text
+        else:
+            d[t.tag] = text
+    return d
```

Comment:
Pretty minor detail really, but it's usually better have a new line at the end of files (do note the red stop sign that github.com adds in the diff). Might be your IDE, I know some of my colleagues had this issue at some point.
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline